### ISSUE TITLE
GET request in kinto-webextensions returns 507 Insufficient Storage

### ISSUE THREAD

**Issue by glasserc** [Kinto/kinto#1225]
```
> http GET 'https://webextensions.settings.services.mozilla.com/v1/buckets/default/collections/ext-Ccc4rbVqgGSk-K7sUJv_68gQdPlmqiqNkffN4uv9-Jc/records?_since=0' Authorization:"Bearer $OAUTH_BEARER_TOKEN"
HTTP/1.1 507 Insufficient Storage
Access-Control-Expose-Headers: Backoff,Retry-After,Alert,Content-Length
Connection: keep-alive
Content-Length: 125
Content-Type: application/json
Date: Mon, 22 May 2017 14:33:04 GMT
X-Content-Type-Options: nosniff

{
    "code": 507,
    "errno": 121,
    "error": "Insufficient Storage",
    "message": "Collection maximum size exceeded (102426 > 102400 Bytes)."
}
```

I can give my bearer token to someone if it would help them track this down.

**Comment by glasserc** [comment]
One theory is that the collection doesn't really exist and is being created by the default bucket, but it shows up in the collections list:

```
> http GET 'https://webextensions.settings.services.mozilla.com/v1/buckets/default/collections' Authorization:"Bearer $OAUTH_BEARER_TOKEN"
HTTP/1.1 200 OK
Access-Control-Expose-Headers: Retry-After, Next-Page, Last-Modified, Pragma, Cache-Control, ETag, Expires, Alert, Backoff, Content-Length, Total-Records
Cache-Control: no-cache, no-store
Connection: keep-alive
Content-Length: 591
Content-Type: application/json
Date: Mon, 22 May 2017 14:39:20 GMT
ETag: "1495226961598"
Last-Modified: Fri, 19 May 2017 20:49:21 GMT
Total-Records: 7
X-Content-Type-Options: nosniff

{
    "data": [
        {
            "id": "ext-kUzzyCxGNwiFvYYi1ZO1MkTOGsoO9jgBZ92WOkbRrJE",
            "last_modified": 1495226961598
        },
        {
            "id": "storage-sync-crypto",
            "last_modified": 1495204273593
        },
        {
            "id": "ext-HiBPc2uTndUKhd4YvUw3v9Uz-tSiSh99_th9vaUlfjg",
            "last_modified": 1493593179857
        },
        {
            "id": "ext-lCLovzdsTMn9wexv1IWrfvv-nl9-zcjqjN6VkzfS_6g",
            "last_modified": 1493593179648
        },
        {
            "id": "ext-Ccc4rbVqgGSk-K7sUJv_68gQdPlmqiqNkffN4uv9-Jc",
            "last_modified": 1493386699683
        },
        {
            "id": "ext-XoxmiYy4pv3TApmYW4H3O8-cyUg4itiSKqDOVseaOC8",
            "last_modified": 1490785806741
        },
        {
            "id": "ext-Y90xoorGgbLBgNeFEapkV_aLf7h4zwd2ns79UeEceYg",
            "last_modified": 1490785480260
        }
    ]
}
```

**Comment by Natim** [comment]
What happens if you delete it and reaccess it?

**Comment by glasserc** [comment]
```
> http DELETE 'https://webextensions.settings.services.mozilla.com/v1/buckets/default/collections/ext-Ccc4rbVqgGSk-K7sUJv_68gQdPlmqiqNkffN4uv9-Jc' Authorization:"Bearer $OAUTH_BEARER_TOKEN"
HTTP/1.1 507 Insufficient Storage
Access-Control-Expose-Headers: Backoff,Retry-After,Alert,Content-Length
Connection: keep-alive
Content-Length: 125
Content-Type: application/json
Date: Mon, 22 May 2017 14:50:07 GMT
X-Content-Type-Options: nosniff

{
    "code": 507,
    "errno": 121,
    "error": "Insufficient Storage",
    "message": "Collection maximum size exceeded (102426 > 102400 Bytes)."
}
```

But this doesn't really surprise me, because we enforce quotas even on DELETE (see e.g. https://github.com/Kinto/kinto/pull/752#r74076173).

**Comment by Natim** [comment]
Does it means that you are in a totally blocked situation?

**Comment by glasserc** [comment]
I can GET some collections, including `storage-sync-crypto`; the ones that I can get are all empty, except for `storage-sync-crypto`, which:

```
> http GET 'https://webextensions.settings.services.mozilla.com/v1/buckets/default/collections/storage-sync-crypto/records?_since=0' Authorization:"Bearer $OAUTH_BEARER_TOKEN"| wc
      0       1    1788
```

... is certainly no bigger than that. I could always delete the bucket; that ought to work, but I don't really want to try that until I have a better idea what happened.

**Comment by Natim** [comment]
Ok can you try to remove another one?

**Comment by Natim** [comment]
Maybe you've reach the max number of collection?

**Comment by glasserc** [comment]
I was able to delete ext-XoxmiYy4pv3TApmYW4H3O8-cyUg4itiSKqDOVseaOC8, which had been one of the empty ones. The collection which produces the 507 still does, with the same number of bytes. Anything worth doing is worth overdoing, so I tried deleting all the other collections too. The ones that I had previously been able to fetch were successfully deleted, but the ones that I hadn't been able to GET before are also impermeable to DELETE. I now have three remaining collections -- storage-sync-crypto and the two I couldn't delete before.
